rubicon/objc/runtime.py

```diff
@@ -981,3 +990,23 @@ def set_ivar(obj, varname, value, weak=False):
         libobjc.object_setIvar(obj, ivar, value)
     else:
         memmove(obj.value + libobjc.ivar_getOffset(ivar), addressof(value), sizeof(vartype))
+
+
+class autoreleasepool:
+    """
+    A context manager that has the same effect as a @autoreleasepool block in Objective-C.
+
+    Any objects that are autoreleased within the context will receive a release message when exiting the context. When
+    running an event loop, AppKit will create an autorelease pool at the beginning of each cycle of the event loop and
+    drain it at the end. You therefore do not need to use @autoreleasepool blocks when running an event loop. However,
+    they may be still be useful when your code temporarily allocates large amounts of memory which you want to
+    explicitly free before the end of a cycle.
+    """
+    def __init__(self):
+        self.pool = None
```

Comment:
Minor thing - there's no need to initialize `self.pool` here, the assignment in `__enter__` is enough. That also guards better against accidentally using the attribute before it's set - trying to use a nonexistant attribute gives an exception, whereas a `None` attribute could be silently passed to the C side as a `NULL` pointer and cause a segfault.